examples/ecs/system_chaining.rs

```diff
@@ -0,0 +1,25 @@
+use anyhow::Result;
+use bevy::prelude::*;
+
+fn main() {
+    App::build()
+        .add_resource(Message("hello".to_string()))
+        .add_system(parse_message_system.chain(handler_system))
+        .run();
+}
+
+struct Message(String);
+
+// this system produces a Result<usize> output by trying to parse the Message resource
+fn parse_message_system(message: Res<Message>) -> Result<usize> {
+    Ok(message.0.parse::<usize>()?)
+}
+
+// This system takes a Result<usize> input and either prints the parsed value or the error message
+// Try changing the Message resource to something that isn't an integer. You should see the error message printed.
```

Comment:
The message is already not an integer. It is `"hello"`.

Reply:
Oops! I changed it to hello to test the error case, then forgot to change it back. Thanks!

Reply:
https://github.com/bevyengine/bevy/commit/db2d20ec1ad89592f0f3c5f9af5f806256cf8a72